### DISCUSSION TITLE
[help] "Right" way of doing a pattern over nested targets?

### DISCUSSION THREAD

**Discussion by lunafazio** [ropensci/targets#1113] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

**Example:**
I have four different model formulas I want to fit some data to. One target maps over the formulas and outputs a list of priors to be tested for each formula. The number of priors I may want to test per formula is not constant or compatible across formulas, so I can't just `cross` them together or `map` over a flattened list. Currently, I just have an intermediate target where I align the indexes for each object and produce the "joint" flattened list.

Having the ability to do something like `map(x, over = y)` would be perfect, but I can imagine there's reasons why this is not available. At any rate, I wonder if there's some nicer (i.e. more automatic) way of achieving this.

**Comment by wlandau** [comment, marked as answer]
From what you are describing, I am picturing a grid of formulas and priors like this:


formula | prior
---|---
formula_1 | prior_1_1
formula_1 | prior_1_2
formula_2 | prior_2_1
formula_2 | prior_2_2
formula_2 | prior_2_3

Once you have this grid in a non-dynamic target, you can map over the rows with a dynamic one. To produce this grid, you could either create it as a `tibble` beforehand, or you could create it with dynamic branching and send it to a non-dynamic target. Constructing formula and prior specifications does not seem to be too computationally demanding, so I would recommend the former approach. Rough sketch:

```r
# _targets.R file
library(targets)
tar_source()

list(
  tar_target(
    name = grid,
    command = get_formula_prior_grid()
  ),
  tar_target(
    name = model,
    command = run_model(formula = grid$formula, prior = grid$prior),
    pattern = map(grid)
  )
)
```

**Reply by lunafazio** [reply]
Ah, good point. There are indeed quite a few simple steps I could take out of the pipeline. It appears the last part of your answer was cut off, but I think I get the gist. Thank you!
